File: Client/src/app/shared/Models/product.ts


export interface IProduct {
    id: number
    name: string
    newPrice: number
    oldPrice: number
    categoryId: number
    categoryName?: string
    trackStock?: boolean
    stockQuantity?: number | null
    photos: string[]
    description: string
    averageRating?: number
    reviewCount?: number
    specifications?: ProductSpecification[]
}

export interface ProductSpecification {
    label: string
    value: string
    sortOrder: number
}

File: ECO.DAL/Entities/Product/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ECO.DAL.Entities.Product
{
    public class Product : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal NewPrice { get; set; }
        public decimal OldPrice { get; set; }
        public bool TrackStock { get; set; }
        public int? StockQuantity { get; set; }
        public decimal? LengthCm { get; set; }
        public decimal? WidthCm { get; set; }
        public decimal? HeightCm { get; set; }
        public decimal? WeightKg { get; set; }
        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public virtual Category Category { get; set; } = null!;
        public virtual List<Photo> Photos { get; set; } = new();
        public virtual List<Review> Reviews { get; set; } = new();
        public virtual List<ProductSpecItem> Specifications { get; set; } = new();
    }
}
